client/src/entrypoints/admin/userbar.js

```diff
@@ -5,249 +5,270 @@
 // This component implements a roving tab index for keyboard navigation
 // Learn more about roving tabIndex: https://w3c.github.io/aria-practices/#kbd_roving_tabindex
```

Comment:
Not for this specific line – I’m finding the JS really hard to review, I’m surprised there are so many changes. In the future I’d have really preferred if we kept the toggle fix and the refactoring completely separate, as:

- It’d have made it easier to review each change separately.
- It’d have made it possible for us to apply the bug fix to both Wagtail’s upcoming version and as a patch to the current release.

Generally, combining multiple unrelated changes just makes it harder for us to get them merged.

I’ll hopefully be able to come back to this later today and make sense of the changes as-is.

Reply:
I committed all the suggestions except for the md file for writing templates - I need a little bit more time to test it.
I also re-opened my PR #9808 for toggle issue bugfix to separate unrelated changes.

Reply:
I tested styling outside the web component, and came up with the solution in my last commit